src/uct/base/uct_mem.c

```diff
@@ -77,30 +78,31 @@ ucs_status_t uct_mem_alloc(void *addr, size_t min_length, unsigned flags,
     ssize_t huge_page_size;
 #endif
 
-    if (min_length == 0) {
+    if (*param->length_p == 0) {
```

Comment:
all accesses to param must be after checking field_mask (in multiple locations in this PR; pls fix all of them)

Reply:
Will address in the next commit.

Reply:
@yosefe I've tried to make sure the param field bits are set before field is used but one assumption I've made is that if PARAM_NAME bit isn't set, then I've used param->name = NULL where applicable. I didn't see checks to make sure that alloc_name is non-NULL, so want to check if this assumption is safe.

Reply:
this is not correct, if field_mask is not specifying some field, it could even not be allocated or accessible in memory